From: Arnd Bergmann <arnd@arndb.de>
Subject: mips: fix setup_zero_pages() prototype
Date: Mon, 4 Dec 2023 12:57:00 +0100

setup_zero_pages() has a local declaration in a platform specific header,
but that is not seen in the file it is defined in:

arch/mips/mm/init.c:60:6: error: no previous prototype for 'setup_zero_pages' [-Werror=missing-prototypes]

Move it to the corresponding global header and include that where needed.

Link: https://lkml.kernel.org/r/20231204115710.2247097-11-arnd@kernel.org
Signed-off-by: Arnd Bergmann <arnd@arndb.de>
Cc: Stephen Rothwell <sfr@rothwell.id.au>
Cc: Thomas Bogendoerfer <tsbogend@alpha.franken.de>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 arch/mips/include/asm/mach-loongson64/mmzone.h |    1 -
 arch/mips/include/asm/mmzone.h                 |    2 ++
 arch/mips/mm/init.c                            |    1 +
 3 files changed, 3 insertions(+), 1 deletion(-)

--- a/arch/mips/include/asm/mach-loongson64/mmzone.h~mips-fix-setup_zero_pages-prototype
+++ a/arch/mips/include/asm/mach-loongson64/mmzone.h
@@ -18,7 +18,6 @@ extern struct pglist_data *__node_data[]
 
 #define NODE_DATA(n)		(__node_data[n])
 
-extern void setup_zero_pages(void);
 extern void __init prom_init_numa_memory(void);
 
 #endif /* _ASM_MACH_MMZONE_H */
--- a/arch/mips/include/asm/mmzone.h~mips-fix-setup_zero_pages-prototype
+++ a/arch/mips/include/asm/mmzone.h
@@ -20,4 +20,6 @@
 #define nid_to_addrbase(nid) 0
 #endif
 
+extern void setup_zero_pages(void);
+
 #endif /* _ASM_MMZONE_H_ */
--- a/arch/mips/mm/init.c~mips-fix-setup_zero_pages-prototype
+++ a/arch/mips/mm/init.c
@@ -38,6 +38,7 @@
 #include <asm/dma.h>
 #include <asm/maar.h>
 #include <asm/mmu_context.h>
+#include <asm/mmzone.h>
 #include <asm/sections.h>
 #include <asm/pgalloc.h>
 #include <asm/tlb.h>
_
